xvec/accessor.py

```diff
@@ -918,6 +918,375 @@ def to_geodataframe(
+        x_axis: int = None,
+        y_axis: int = None,
+        stat: str = "mean",
+        dask: bool = False,
+    ):
+        """Aggregate the values from a dataset over a polygon geometry.
+
+        The CRS of the raster and that of points need to be in wgs84.
+        Xvec does not verify their equality.
+
+        Parameters
+        ----------
+        geom : Polygon[shapely.Geometry]
+            An arrray-like (1-D) of shapely geometry, like a numpy array or GeoPandas
+            GeoSeries.
+        trans : affine.Affine
+            Affine transformer.
+            Representing the geometric transformation applied to the data.
+        x_coords : Hashable
+            Name of the axis containing ``x`` coordinates.
+        y_coords : Hashable
+            Name of the axis containing ``y`` coordinates.
+        x_axis : int
+            Axis order of  ``x`` coordinates.
+        y_axis : int
+            Axis order of  ``y`` coordinates.
+        var : Hashable
+            Name of the variable in the dataset to aggregate its values.
+        stat : Hashable
+            Spatial aggregation statistic method, by default "mean". It supports the
+            following statistcs: ['mean', 'median', 'min', 'max', 'sum']
+        dask : bool,
+            If the input is dask array or not.
+
+        Returns
+        -------
+        Array
+            Aggregated values over the geometry.
+
+        """
+        try:
+            import geopandas as gpd
+        except ImportError as err:
+            raise ImportError(
+                "The geopandas package is required for `xvec.agg_geom()`. "
+                "You can install it using 'conda install -c conda-forge geopandas' or "
+                "'pip install geopandas'."
+            ) from err
+
+        try:
+            import rasterio
+        except ImportError as err:
+            raise ImportError(
+                "The rasterio package is required for `xvec.agg_geom()`. "
+                "You can install it using 'conda install -c conda-forge rasterio' or "
+                "'pip install rasterio'."
+            ) from err
+
+        try:
+            import gc
+        except ImportError as err:
+            raise ImportError(
+                "The gc package is required for `xvec.agg_geom()`. "
+                "Make sure 'gc' is included in the standard library"
+                "Check your Python installation for any issues."
+            ) from err
+
+        # Create a GeoSeries from the geometry
+        geo_series = gpd.GeoSeries(geom)
+
+        # Convert the GeoSeries to a GeometryArray
+        geometry_array = geo_series.geometry.array
+
+        xar_chunk = self._obj[var]
+        mask = rasterio.features.geometry_mask(
+            geometry_array,
+            out_shape=(xar_chunk.shape[0], xar_chunk.shape[1]),
+            transform=trans,
+        )
+
+        masked_data = xar_chunk * mask[:, :, np.newaxis]
+        del mask, xar_chunk
+        gc.collect()
```

Comment:
How important this is in your opinion?

Reply:
It's important in case of limited resources. I've tested it over continental level "sentinel-2 tiles" for 150 polygons and the code crashed without it. It helps to save memory when have big data.

Reply:
Okay!